docs/Design/Alpha/Data Model.md

```diff
@@ -0,0 +1,215 @@
+        int id PK 
+        string(255) url
+        int artist_id FK
+    }
+    
+    artist }o--|{ user : "artist_manager"
+    artist }|--o{ album : "artist_album"
+    artist }o--o{ tag : "tag_artist"
+    artist ||--o{ link : "" 
+    artist }o--o{ artist : "artist_members"
+    album }|--|{ track : "album_track"
+    album }o--o{ tag : "tag_album"
+    track }|--o{ artist : "track_artist"
+    track }o--o{ tag : "tag_track"
+    playlist ||--o{ playlist_track : ""
+    playlist_track }o--|| track : ""
+    user }o--o{ artist : "user_following"
+    user }o--o{ album : "user_collection_album"
+    user }o--o{ track : "user_collection_track"
+    user ||--o{ playlist : ""
+    user_purchase_album }o--|| album : ""
+    user_purchase_album }o--|| user : ""
+    user_purchase_track }o--|| track : ""
+    user_purchase_track }o--|| user : ""
+```
+
+## Notes
+
+- While most of the many-to-many relationship tables are minimally simple (two columns, each containing IDs),
+  there are a few (`playlist_track`, `user_purchase_album`, `user_purchase_track`) that are
+  defined explicitly within the diagram because they are a touch more complex.
+
+### Artists
+- If an **Artist** is a group, it can be linked to individual **Artist**s as members so that **User**s
+can browse the **Artist**'s solo work or work with other groups. **User**s will also be able to browse 
+the groups of which an **Artist** is a member from the individual **Artist**'s page.
+- An **Artist** may be without a manager if a **User** uploads an **Album** and credits an artist who
+doesn't have an account. A dummy **Artist** entity would be created and would somehow have to be claimed
+by the real artist in the event they do decide to make an account.
+
+### Album
+- Allows **Artist**s to specify a `release_year` if the exact `release_date` is unknown
+
+### Tracks
+- The price for a **Track** is nullable in case the **Artist** doesn't want to allow individual **Track**s
+to be purchased. If the **Artist** wants the **Track** to be free, the price can be set to zero.
+- `is_streamable` allows users to stream a **Track** for free even if it is available for purchase
+- **Track**s have their own relationship to **Artist**s in the case of a guest **Artist** or other contributor
+who is not credited on the full album. If a **Track** does not relate to any **Artist**s, use the **Artist**s 
+related to the **Track**'s **Album**.
+
+### Links
+- **Artist**s can display links on their page to social media, merch, etc.
+
+### User
+- The entity described by the diagram is the most basic data required to represent a **User**. 
+The specification is likely to change drastically during the design for authentication and 
+authorization.
+
+### Purchase
+- There are separate tables for **Album** and **Track** purchases so that the database can reliably
+track the relationships between the relevant entities.
+- The **Purchase** tables may change once we figure out our e-commerce platform.
```

Comment:
Commenting here instead of the ER documentation because of the second point raised in this paragraph.
Currently other music platforms support purchasing music without logging in. The platforms will accept ecommerce and an email address and then send the purchased music to the email address. We may want to adjust (or note within it) our data model in order to support this functionality.

Reply:
I think this can be a future task. Tying things to users will be overall simpler, but anonymous purchase would be nice to have. It's probably something we want to have before a full release to the public, which is way down the road.